lib/spack/spack/config.py

```diff
@@ -539,13 +549,25 @@ def override(path_or_scope, value=None):
     an internal config scope for it and push/pop that scope.
 
     """
+    base_name = 'overrides-'
     if isinstance(path_or_scope, ConfigScope):
         overrides = path_or_scope
         config.push_scope(path_or_scope)
     else:
-        overrides = InternalConfigScope('overrides')
+        # Ensure the new override gets a unique scope name
+        current_overrides = [s.name for s in
+                             config.matching_scopes(r'^{0}'.format(base_name))]
```

Comment:
I think we could do this without regex more efficiently. Not worth changing it, I don't think this code is any sort of performance bottleneck.

Reply:
True.  I could use `startswith` as I did in the unit tests, for example.  IIRC I think`regex` came  up when I told Todd (tgamblin) what I was planning.

Reply:
BTW. The aforementioned discussion was the impetus for adding `spack.config.matching_scopes`.